### ISSUE TITLE
What's the result about half KG in metaQA? 

### ISSUE THREAD

**Issue by ToneLi** [malllabiisc/EmbedKGQA#40]
By using your code without your relation matching in half KG metaQA, I found the result is 0.68 in hop1, 0.51 in hop2, 0.45 in hop3, there is a big gap with your paper report (even without relation matching, relation matching is used to solve: the big distance between topic entity and answers). I do not undertand how to achieve your relation matching in half KG, can you report your result without relation matching? By your describtion, it's not easy to realise the relation matching model.

**Comment by apoorvumang** [comment]
This is on half KG or full KG?

**Comment by apoorvumang** [comment]
We did not use relation matching for half KG. Can you give the exact commands you used? Did you use the pretrained KG Embeddings or trained them again?

**Comment by ToneLi** [comment]
OK, thanks! I used the pretrained KG ebmeddings you provided, and I train them again, I find the hit@1 is just 0.2. These embedding I all try, but result is not good, .....

my command,
python3 main.py --mode train --relation_dim 200 --hidden_dim 256 \
--gpu 2 --freeze 0 --batch_size 128 --validate_every 5 --hops 2 --lr 0.0005 --entdrop 0.1 --reldrop 0.2  --scoredrop 0.2 \
--decay 1.0 --model ComplEx --patience 5 --ls 0.0 --kg_type half

**Comment by ToneLi** [comment]
I think the good or bad about pretrained embedding is very important, I used your code and data in metaQA half KG, get the 0.2 hit@1 while full kg is 1.0 hit@1, so the result in half kg is too low, can you tell me the result about hit@1 in metaQA half KG?

**Comment by apoorvumang** [comment]
> OK, thanks! I used the pretrained KG ebmeddings you provided, and I train them again,

Do you mean you are using pretrained embeddings or training them again? For pretrained, I am able to reproduce the following:

```
python3 main.py --mode train --relation_dim 200 --hidden_dim 256 \
--gpu 4 --freeze 0 --batch_size 256 --validate_every 5 --hops 1 --kg_type half --lr 0.001 --entdrop 0.2 --reldrop 0.3 --scoredrop 0.3 \
--decay 1.0 --model ComplEx --patience 5 --ls 0.1 --l3_reg 0.0001
```
0.82 hit@1 within 10 epochs (still not converged)

![image](https://user-images.githubusercontent.com/1957903/96011248-4f004880-0e60-11eb-9e38-63a4123be17c.png)

Similarly, for 2 hop (which takes longer to converge), I got the following within 5 epochs. It will take ~30 epochs to reach accuracy reported in paper but it's definitely not 0.2

```
python3 main.py --mode train --relation_dim 200 --hidden_dim 256 \
--gpu 3 --freeze 0 --batch_size 256 --validate_every 5 --hops 2 --kg_type half --lr 0.001 --entdrop 0.2 --reldrop 0.3 --scoredrop 0.3 \
--decay 1.0 --model ComplEx --patience 5 --ls 0.1 --l3_reg 0.0001
```
![image](https://user-images.githubusercontent.com/1957903/96012121-6d1a7880-0e61-11eb-88c4-e1a69ca4876e.png)

Maybe you could try by cloning a fresh copy of the repo and try the above commands? Or if you are retraining the KG embeddings, could you give me the commands you are using along with how you are creating the embedding files from the model checkpoint?

**Comment by ToneLi** [comment]
OK, I think I found the reason,  my parameters is not same with you! Thanks for your reply!!  If I have probems again, I will contact you again! Thanks!

**Comment by ToneLi** [comment]
By the way, can you apply the repretrained half KG model  file in websqp and command, just like above?

**Comment by apoorvumang** [comment]
> By the way, can you apply the repretrained half KG model file in websqp and command, just like above?

Yes I think you should be able to do that
